lib/webpacker/source.rb

```diff
@@ -1,38 +1,30 @@
-# Translates a logical reference for a pack source into the final path needed in the HTML.
-# This translation takes into account whether digesting is configured to happen, which it
-# is by default in the production environment (as set via
-# `Rails.configuration.x.webpacker[:digesting] = true`).
+require "webpacker/manifest"
+require "webpacker/dev_server"
+
+# Translates a logical reference for a pack source into the final
+# path needed in the HTML using generated manifest.json file.
 class Webpacker::Source
-  def initialize(name)
-    @name = name
+  SOURCE_EXTENSIONS = { javascript: ".js", stylesheet: ".css" }.freeze
+
+  def initialize(name, options = {})
+    @name    = name
+    @options = options
   end
 
   def path
-    if config[:dev_server_host].present?
-      "#{config[:dev_server_host]}/#{filename}"
-    elsif config[:digesting]
-      File.join(dist_path, digested_filename)
+    if Webpacker::DevServer.running?
+      Webpacker::DevServer.resolve(compute_source_with_extname)
     else
-      File.join(dist_path, filename)
+      Webpacker::Manifest.lookup(compute_source_with_extname)
     end
   end
 
   private
-    attr_accessor :name
-
-    def config
-      Rails.configuration.x.webpacker
-    end
-
-    def digested_filename
-      Webpacker::Digests.lookup(name)
-    end
-
-    def dist_path
-      config[:packs_dist_path]
-    end
+    attr_accessor :name, :options
 
-    def filename
-      "#{name}.js"
+    def compute_source_with_extname
+      extname = SOURCE_EXTENSIONS[options[:type]]
+      return name + extname if extname && File.extname(name) != extname
+      name
```

Comment:
Could `ActionView::Helpers::AssetUrlHelper.compute_asset_extname` be used to avoid duplicating this logic with extensions?

Something like
```
def compute_source_with_extname
  return "#{name}#{compute_asset_extname(source, options)}"
end
```